File: v1/implementation/Tumbuhan.cpp
#include "../header/Tumbuhan.h"
#include "../header/Point.h"
#include <thread>
#include <chrono>
using namespace std;

Tumbuhan::Tumbuhan(const Point& P) : Makhluk(TUMBUHAN_ID, TUMBUHAN_MAXAGE)
{
	status = 1;
	age = 0;
	pos = P;
}

File: TubesMakhluk/src/makhluk/Tumbuhan.java
package makhluk;

import point.Point;

/** {@inheritdoc} Kelas Tumbuhan adalah kelas turunan dari Makhluk 
 *  yang merepresentasikan sebuah tumbuhan.
 *  @author Robert Sebastian Herlim  / 13514061
 *  @version 1.0
 */
public class Tumbuhan extends Makhluk {
    /** Atribut ID tumbuhan.
     */
    private static final char TUMBUHAN_ID = 'G';
    /** Atribut umur maksimal tumbuhan.
     */
    private static final int TUMBUHAN_MAXAGE = 300;
    /** Menciptakan sebuah tumbuhan di koordinat p.
     * @param p koordinat penciptaan tumbuhan
     */
    public Tumbuhan(final Point p) {
        super(TUMBUHAN_ID, TUMBUHAN_MAXAGE);
        age = 0;
        status = 1;
        pos = p;
	}
}
